### ISSUE TITLE
System.IO.FileNotFoundException: Could not load file or assembly 'System.Net.Security ...' ocurring in Windows using rc1-final and OSX using rc1-update1

### ISSUE THREAD

**Issue by mogarick** [dotnet/corefx#5858]
### TLDR; 
I'm getting a runtime error only when building in **OSX** with `dnx-coreclr-x64-1.0.0-rc1-update1` or building in **Windows** with `dnx-coreclr-win-x64.1.0.0-rc1-final` but if I build in **Windows** with `dnx-coreclr-win-x64.1.0.0-rc1-update1` the error doesn't happen. Isn't it supposed that if you use the same version number for both platforms is because both of them are the same and have the same features and fixes? If so, is it possible there was a build mistake and the **OSX** version released is not really `rc1-update1` **(CoreClr-x64-1.0.0-rc1-16231)** but `rc1-final` **(CoreClr-x64-1.0.0-rc1-16202)**. Here's is the error trace:

```
An unhandled exception has occurred: Could not load file or assembly 'System.Net.Security, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

System.IO.FileNotFoundException: Could not load file or assembly 'System.Net.Security, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

File name: 'System.Net.Security, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' ---> System.IO.FileNotFoundException: Could not load the specified file.
File name: 'System.Net.Security'
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyName(AssemblyName assemblyName)
   at System.Runtime.Loader.AssemblyLoadContext.Resolve(IntPtr gchManagedAssemblyLoadContext, AssemblyName assemblyName)
   at MailKit.Net.Smtp.SmtpClient.Connect(String host, Int32 port, SecureSocketOptions options, CancellationToken cancellationToken)
   at MailKit.MailService.Connect(String host, Int32 port, Boolean useSsl, CancellationToken cancellationToken)
   at GReembolsos.Controllers.GastosController.EnviarEmail()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.AspNet.Mvc.Controllers.ControllerActionExecutor.ExecuteAsync(MethodInfo actionMethodInfo, Object instance, Object[] orderedActionArguments)
   at Microsoft.AspNet.Mvc.Controllers.ControllerActionInvoker.<InvokeActionAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNet.Mvc.Controllers.FilterActionInvoker.<InvokeActionFilterAsync>d__53.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNet.Mvc.Controllers.FilterActionInvoker.<InvokeAsync>d__44.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNet.Mvc.Infrastructure.MvcRouteHandler.<RouteAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at Microsoft.AspNet.Mvc.Routing.InnerAttributeRoute.<RouteAsync>d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at Microsoft.AspNet.Routing.RouteCollection.<RouteAsync>d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNet.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNet.IISPlatformHandler.IISPlatformHandlerMiddleware.<Invoke>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNet.Diagnostics.ExceptionHandlerMiddleware.<Invoke>d__6.MoveNext()
```
### Update
I've just found issue #2483 and certainly appears to be the reason of my error but I'm not an expert on the subject and if it is the case then my questions / doubts are still valid I think. 
Thank you in advance for any help.

### Long Story
I'm working on a prototype using **Mac OSX** and **Visual Studio Code** with `dnxcore50` using `coreclr-rc1-update1` but I'm getting the runtime error when trying to send email using **MailKit** lib. @jstedfast, the lib developer said the lib was working perfectly fine when things ran on windows so the problem could be in **the CLR version** for non Windows environments.
I setup a **Windows 10** VM to verify if the problem was indeed not present. The test was successful. No runtime error was thrown.
I tried to deploy as an **Azure Web Site**, and after a few hours of troubleshooting trying to make **coreclr** and not **clr** the runtime target I finally got the app built. But the test failed with the same error I was getting in **Mac OSX** even though it was a Windows Environment! 

I made more research and found this:
* In my local **Windows 10** when I built the project the **DNU** starting message was `Microsoft .NET Development Utility CoreClr-x64-1.0.0-rc1-16231`
* In the **Azure** building log stream the **DNU** starting message was `Microsoft .NET Development Utility CoreClr-x64-1.0.0-rc1-16202`

Looking at the `deploy.cmd` that I had previously modified to target **coreclr** and not **clr** I noticed that the targeted runtime was `dnx-coreclr-win-x64.1.0.0-rc1-final` In my **Windows 10** local the target runtime was `dnx-coreclr-win-x64.1.0.0-rc1-update1`

So I changed the target runtime in `deploy.cmd` to `dnx-coreclr-win-x64.1.0.0-rc1-update1`, pushed to **Azure** and now in the building log stream the **DNU** starting message was `Microsoft .NET Development Utility CoreClr-x64-1.0.0-rc1-16231`, the same as my **Windows 10** local. I ran my test again and the error was gone!! 

After that I concluded that whatever the root cause of the exception I was getting, the problem appears to be fixed in `rc1-update1`, BUT...
it turns out that in **OSX** the error is triggering even though the **DNU** starting message is `Microsoft .NET Development Utility CoreClr-x64-1.0.0-rc1-16231` when I build the project.

So it makes me think it could be that the **OSX** version is not really `rc1-update1` **(CoreClr-x64-1.0.0-rc1-16231)** but `rc1-final` **(CoreClr-x64-1.0.0-rc1-16202)**. It's just a though as I don't know if it's possible or correct/normal  that the same version on both platforms can have different features / fixes. That's why I'm submitting this issue.

**Comment by joshfree** [comment]
@ericeil can you please help out here?
/cc @vijaykota

**Comment by ericeil** [comment]
@mogarick, from your description it looks to me like this is, as you said, a case of functionality that's simply not yet done on OSX.  Certainly once we ship the final 1.0 release, this functionality should be everywhere, but there's work yet to do.  I'll close this as a duplicate of #2483.

**Comment by ericeil** [comment]
Looks like I may have jumped on this a little too quickly....reopening.

**Comment by ericeil** [comment]
I've verified that I can use System.Net.Security.SslStream from a simple project, on OSX running under DNX/CoreCLR RC2 (dnu version string CoreClr-x64-1.0.0-rc2-16357).  Please let me know if moving forward to a more recent version does not resolve this for you.

**Comment by mogarick** [comment]
@ericeil Where can I download RC2?

**Comment by ericeil** [comment]
I think I did this:

```
dnvm upgrade -r coreclr -u
```
